src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs

```diff
@@ -1531,5 +1533,53 @@ private static string GetDscValue(int id, string value)
                     return value;
             }
         }
+
+        public void CreateNewCredential(NewS3CredentialRequestParams obj)
+        {
+            using (var dataContainer = CreateDatabaseDataContainer(obj.ConnectionUri, true, String.Empty))
+            {
+                SecureString secret = $"{obj.Secret}".StringToSecureString();
+                secret.MakeReadOnly();
+                string identity = "S3 Access Key";
+                Credential credential = new Credential(dataContainer.Server, obj.S3Url)
+                {
+                    Identity = identity
+                };
+                credential.Create(identity, secret);
+            }
+        }
+
+        /// <summary>
+        /// Get S3 credentials in the current connection
+        /// </summary>
+        /// <param name="connectionUri">The connection uri</param>
+        /// <returns>Returns the list of credentials names</returns>
+        public List<string> GetS3Credentials(string connectionUri)
+        {
+            List<string> credentials = new List<string>();
+            ConnectionInfo connectionInfo = this.GetConnectionInfo(connectionUri);
+            using (SqlConnection sqlConn = ConnectionService.OpenSqlConnection(connectionInfo))
+            {
+                if (sqlConn != null)
+                {
+                    using (var cmd = new SqlCommand { Connection = sqlConn })
+                    {
+                        cmd.CommandText = "SELECT [NAME] FROM sys.credentials";
+                        cmd.ExecuteNonQuery();
+                        using (IDataReader reader = cmd.ExecuteReader())
+                        {
+                            while (reader.Read())
+                            {
+                                if(reader.GetString(0).StartsWith("s3"))
```

Comment:
1. The casing doesn't match the credential you create above, so is that going to work correctly to find it?
2. According to https://learn.microsoft.com/en-us/sql/relational-databases/backup-restore/sql-server-backup-to-url-s3-compatible-object-storage?view=sql-server-ver16#create-credential the identity must always exactly be `S3 Access Key`, so what about searching for that exactly?

Reply:
yup, I added this in the ADS, but using the identity part makes more sense. I'll change it to that instead